repository/src/main/java/org/alfresco/repo/download/ZipDownloadExporter.java

```diff
@@ -1,329 +1,344 @@
+            DownloadStorage downloadStorage, DictionaryService dictionaryService,
+            NodeRef downloadNodeRef, long total, long totalFileCount)
+    {
+        super(checkOutCheckInService, nodeService);
+        try
+        {
+            this.outputStream = new FileOutputStream(zipFile);
+            this.updateService = updateService;
+            this.transactionHelper = transactionHelper;
+            this.downloadStorage = downloadStorage;
+            this.dictionaryService = dictionaryService;
+
+            this.downloadNodeRef = downloadNodeRef;
+            this.total = total;
+            this.totalFileCount = totalFileCount;
+        }
+        catch (FileNotFoundException e)
+        {
+            throw new ExporterException("Failed to create zip file", e);
+        }
+    }
+
+    @Override
+    public void start(final ExporterContext context)
+    {
+        zipStream = new ZipArchiveOutputStream(outputStream);
+        // NOTE: This encoding allows us to workaround bug...
+        // http://bugs.sun.com/bugdatabase/view_bug.do;:WuuT?bug_id=4820807
+        zipStream.setEncoding("UTF-8");
+        zipStream.setCreateUnicodeExtraFields(UnicodeExtraFieldPolicy.ALWAYS);
+        zipStream.setUseLanguageEncodingFlag(true);
+        zipStream.setFallbackToUTF8(true);
+    }
+
+    @Override
+    public void startNode(NodeRef nodeRef)
+    {
+        this.currentName = (String) nodeService.getProperty(nodeRef, ContentModel.PROP_NAME);
+        this.zipTimestampCreated = (Date) nodeService.getProperty(nodeRef, ContentModel.PROP_CREATED);
+        this.zipTimestampModified = (Date) nodeService.getProperty(nodeRef, ContentModel.PROP_MODIFIED);
+        path.push(new Pair<String, NodeRef>(currentName, nodeRef));
+        if (dictionaryService.isSubClass(nodeService.getType(nodeRef), ContentModel.TYPE_FOLDER))
+        {
+            String path = getPath() + PATH_SEPARATOR;
+            ZipArchiveEntry archiveEntry = new ZipArchiveEntry(path);
+            try
+            {
+                archiveEntry.setTime(zipTimestampCreated.getTime());
+                archiveEntry.setCreationTime(FileTime.fromMillis(zipTimestampCreated.getTime()));
+                archiveEntry.setLastModifiedTime(FileTime.fromMillis(zipTimestampModified.getTime()));
+                zipStream.putArchiveEntry(archiveEntry);
+                zipStream.closeArchiveEntry();
+            }
+            catch (IOException e)
+            {
+                throw new ExporterException("Unexpected IOException adding folder entry", e);
+            }
+        }
+    }
+
+    @Override
+    public void contentImpl(NodeRef nodeRef, QName property, InputStream content, ContentData contentData, int index)
+    {
+        // if the content stream to output is empty, then just return content descriptor as is
+        if (content == null)
+        {
+            log.info("Archiving content has been removed or modified for the specified NodeReference: " + nodeRef
```

Comment:
There's a logger, which has been configured for which we are getting content data as null!